src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,269 @@
+    );
+
+    const expander = nodes[currentIndex].querySelector(
+      '.iui-tree-node-content-expander-icon',
+    ) as HTMLElement;
+
+    switch (event.key) {
+      case 'ArrowDown': {
+        newIndex = currentIndex + 1;
+        break;
+      }
+      case 'ArrowUp': {
+        newIndex = currentIndex - 1;
+        break;
+      }
+      case 'ArrowLeft':
+        if (
+          expander === null ||
+          !expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          let parentIndex = -1;
+          for (let n = 0; n < nodes.length; n++) {
+            const parentOwns = (nodes[n].parentElement
+              ?.lastChild as HTMLElement)
+              .getAttribute('aria-owns')
+              ?.split(', ');
+            if (
+              parentOwns &&
+              parentOwns?.findIndex(
+                (id) => id === nodes[currentIndex].parentElement?.id,
+              ) != -1
+            ) {
+              parentIndex = n;
+              break;
+            }
+          }
+          if (parentIndex != -1) {
+            newIndex = parentIndex;
+          } else {
+            newIndex = currentIndex - 1;
+          }
+        } else {
+          expander.parentElement?.click();
+        }
+        break;
+      case 'ArrowRight':
+        if (
+          expander === null ||
+          expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          newIndex = currentIndex + 1;
+        } else {
+          expander.parentElement?.click();
+          return;
+        }
+        break;
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        nodes[currentIndex].click();
+        event.preventDefault();
+        return;
+    }
+
+    if (newIndex >= 0 && newIndex < nodes.length) {
+      nodes[newIndex].focus();
+      event.preventDefault();
+    }
+  };
+
+  const flatNodesList = React.useMemo(() => {
+    const flatList: FlatNode<T>[] = [];
+
+    const flatNodes = (nodes: T[] = [], depth = 0) => {
+      const nodeIdList = Array<string>();
+      nodes.forEach((element) => {
+        const flatNode: FlatNode<T> = {
+          nodeData: getNode(element),
+          depth: depth,
+        };
+        nodeIdList.push(flatNode.nodeData.nodeId);
+        flatList.push(flatNode);
+        if (flatNode.nodeData.isExpanded) {
+          const subNodeIds = flatNodes(flatNode.nodeData.subNodes, depth + 1);
+          flatNode.subNodeIds = subNodeIds;
+        }
+      });
+      return nodeIdList;
+    };
+    flatNodes(data);
+    return flatList;
+  }, [data, getNode]);
+
+  return (
+    <ul
+      className={cx('iui-tree', className)}
+      role='tree'
+      onKeyDown={handleKeyDown}
+      ref={treeRef}
+      {...rest}
+    >
+      {flatNodesList.map((flatNode) => (
+        <TreeContext.Provider
+          key={flatNode.nodeData.nodeId}
+          value={{
```

Comment:
There is errors in console:
![image](https://user-images.githubusercontent.com/36186912/150942804-068095b1-0eb1-4383-b317-8e3509a7d061.png)
We are passing these using context but they are still in `nodeRenderer` props and in story we use `...rest` to pass remaining props.
@veekeys  @mayank99 Should we just enlist all props in our stories or should we do a hack like in Table:
![image](https://user-images.githubusercontent.com/36186912/150944762-394dd367-eb10-4172-b0f9-1344cba1c4aa.png)

Reply:
wait, why rest gives errors? We pass rest on TreeNode component, which has those props, no?

Reply:
It doesn't have these 2 currently.

Reply:
Why are they in `nodeRenderer` if they're passed using context?

I don't think we can expect users to manually list those props.

Reply:
I added `hasSubNodes` as prop to `TreeNode` so that warning is gone now. 
But I'm not sure how to fix the warning with `subNodes` since it is not needed as a prop in `TreeNode`

Reply:
Fixed console error